tests/test_state/test_storage_state.py

```diff
@@ -36,6 +57,128 @@ def test_market_cycle():
             assert storage_state.offered_buy_kWh[future_time_slots[0]] == 0
             assert storage_state.offered_sell_kWh[future_time_slots[0]] == 0
 
+    def test_market_cycle_update_used_storage(self):
+        storage_state = StorageState(initial_soc=100,
+                                     capacity=100)
+        past_time_slot, current_time_slot, future_time_slots = self._initialize_time_slots()
+        active_market_slot_time_list = [past_time_slot, current_time_slot, *future_time_slots]
+        storage_state.add_default_values_to_state_profiles(active_market_slot_time_list)
+        storage_state.pledged_sell_kWh[past_time_slot] = 10
+        storage_state.pledged_buy_kWh[past_time_slot] = 0
+        storage_state.market_cycle(past_time_slot, current_time_slot, future_time_slots)
+        assert storage_state.used_storage == 90
+        storage_state.pledged_sell_kWh[past_time_slot] = 0
+        storage_state.pledged_buy_kWh[past_time_slot] = 10
+        storage_state.market_cycle(past_time_slot, current_time_slot, future_time_slots)
+        assert storage_state.used_storage == 100
+
+    def test_market_cycle_ess_share_time_series_dict(self):
+        storage_state = StorageState(initial_soc=100,
+                                     capacity=100,
+                                     initial_energy_origin=ESSEnergyOrigin.LOCAL,
+                                     max_abs_battery_power_kW=15)
+        past_time_slot, current_time_slot, future_time_slots = self._initialize_time_slots()
+        active_market_slot_time_list = [past_time_slot, current_time_slot, *future_time_slots]
+        storage_state.add_default_values_to_state_profiles(active_market_slot_time_list)
+        energy = 0.3
+        storage_state.activate(
+            slot_length=duration(minutes=15), current_time_slot=current_time_slot)
+        storage_state.register_energy_from_one_sided_market_accept_offer(
+            energy, past_time_slot, ESSEnergyOrigin.LOCAL)
+        storage_state.register_energy_from_one_sided_market_accept_offer(
+            energy, past_time_slot, ESSEnergyOrigin.UNKNOWN)
+        storage_state.register_energy_from_one_sided_market_accept_offer(
+            energy, past_time_slot, ESSEnergyOrigin.UNKNOWN)
+        storage_state.market_cycle(past_time_slot, current_time_slot, future_time_slots)
+        expected_time_series = {ESSEnergyOrigin.LOCAL: storage_state.initial_capacity_kWh + energy,
```

Comment:
This is not a time series, only the value for the expected past_time_slot, right?
```suggestion
        expected_ess_share_last_market = {ESSEnergyOrigin.LOCAL: storage_state.initial_capacity_kWh + energy,
```